### PR TITLE
[Fix][Zeta] If Zeta not a TCP discovery, it cannot find other members

### LINKED ISSUE #7754: [Bug] [Zeta] Zeta Server cannot find other members via Kubernetes Discovery
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/seatunnel/issues?q=is%3Aissue+label%3A%22bug%22) and found no similar issues.

### What happened

Please see #7750 

### SeaTunnel Version

2.3.7

### SeaTunnel Config

```conf
seatunnel:
  engine:
    history-job-expire-minutes: 1440
    backup-count: 1
    queue-type: blockingqueue
    print-execution-info-interval: 60
    print-job-metrics-info-interval: 60
    classloader-cache-mode: true
    slot-service:
      dynamic-slot: true
    checkpoint:
      interval: 10000
      timeout: 9000000
      storage:
        type: hdfs
        max-retained: 3
        plugin-config:
          namespace: /tmp/seatunnel/checkpoint_snapshot
          storage.type: hdfs
          fs.defaultFS: file:///tmp/
```

### Running Command

```shell
/opt/seatunnel/bin/seatunnel-cluster.sh
```

### Error Exception

```log
No
```

### Zeta or Flink or Spark Version

2.3.7

### Java or Scala Version

_No response_

### Screenshots

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by loustler**
### Purpose of this pull request
Fix a bug in SeaTunnelNodeContext, please see #7750.

It resolved #7754 

### Does this PR introduce _any_ user-facing change?
NO

### How was this patch tested?
Packaged the sea tunnel-starter.jar, place it it in a docker image

### Check list

* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update [plugin-mapping.properties](https://github.com/apache/seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  2. Update the pom file of [seatunnel-dist](https://github.com/apache/seatunnel/blob/dev/seatunnel-dist/pom.xml)
  3. Add ci label in [label-scope-conf](https://github.com/apache/seatunnel/blob/dev/.github/workflows/labeler/label-scope-conf.yml)
  4. Add e2e testcase in [seatunnel-e2e](https://github.com/apache/seatunnel/tree/dev/seatunnel-e2e/seatunnel-connector-v2-e2e/)
  5. Update connector [plugin_config](https://github.com/apache/seatunnel/blob/dev/config/plugin_config)
* [ ] Update the [`release-note`](https://github.com/apache/seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by loustler** [comment]
@Hisoka-X @liunaijie
I'll add a test case for it later

**Comment by loustler** [comment]
I'll refactor this a little bit, please wait

**Review by Hisoka-X — COMMENTED** [review]
Thanks @loustler ! Waiting test case.

**Comment by loustler** [comment]
@Hisoka-X @liunaijie 
A new test case is passed.
Please review this

File: seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/custom_config/hazelcast-kubernetes-discovery.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  cluster-name: seatunnel
+  network:
+    rest-api:
+      enabled: true
+      endpoint-groups:
+        CLUSTER_WRITE:
+          enabled: true
+        DATA:
+          enabled: true
+    join:
```
Comment (Hisoka-X) [inline seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/custom_config/hazelcast-kubernetes-discovery.yaml:28]: how about share this way to other users in the doc？

Comment (Hisoka-X) [reply seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/custom_config/hazelcast-kubernetes-discovery.yaml:28]: Forget it. Use TCP is a better way. WDYT? @loustler

Comment (loustler) [reply seatunnel-e2e/seatunnel-engine-e2e/seatunnel-engine-k8s-e2e/src/test/resources/custom_config/hazelcast-kubernetes-discovery.yaml:28]: Yes. A TCP discovery is a better and stable way, and it's a recommendation by HazelCast

**Review by Hisoka-X — APPROVED** [review]
Thanks @loustler!

**Review by liunaijie — APPROVED** [review]

**Pull request merged**